pkg/daemon/daemon.go

```diff
@@ -1290,12 +1294,67 @@ func (dn *Daemon) completeUpdate(desiredConfigName string) error {
 		return err
 	}
 
+	if err := dn.removeUpdateInProgressTaint(); err != nil {
+		return err
+	}
+
 	dn.logSystem("Update completed for config %s and node has been successfully uncordoned", desiredConfigName)
 	dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "Uncordon", fmt.Sprintf("Update completed for config %s and node has been uncordoned", desiredConfigName))
 
 	return nil
 }
 
+func (dn *Daemon) removeUpdateInProgressTaint() error {
+	// TODO: Move these 2 vars to common location to be shared with
+	var (
+		nodeUpdateBackoff = wait.Backoff{
+			Steps:    5,
+			Duration: 100 * time.Millisecond,
+			Jitter:   1.0,
+		}
+		// NodeUpdateTaint is a taint applied by MCC when the update of node starts.
+		NodeUpdateTaint = &corev1.Taint{
+			Key:    "UpdateInProgress",
+			Effect: corev1.TaintEffectPreferNoSchedule,
+		}
+	)
+	return clientretry.RetryOnConflict(nodeUpdateBackoff, func() error {
+		oldNode, err := dn.kubeClient.CoreV1().Nodes().Get(context.TODO(), dn.name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		oldData, err := json.Marshal(oldNode)
+		if err != nil {
+			return err
+		}
+
+		newNode := oldNode.DeepCopy()
+		if newNode.Spec.Taints == nil {
+			newNode.Spec.Taints = []corev1.Taint{}
+		}
+
```

Comment:
I think I'm tired but can we get a comment here on this loop thru append are doing bc future me will also be tired and easily confused ;)

Reply:
I just want to remove a taint from the node. I can add a comment. Technically, I can convert this to set so that deletion will be easier but I don't think the current implementation is going to be that expensive considering the number of taints we have.

Reply:
I think it's fine to leave implementation as-is now, just add comment :+1: 

we can refactor if necessary later